metals/src/main/scala/scala/meta/internal/metals/formatting/IndentOnPaste.scala

```diff
@@ -77,20 +71,14 @@ object IndentOnPaste extends RangeFormatter {
       else
         FmtOptions.tabs
 
-    val pastedLines =
-      splitLines
-        .slice(startLine, endLine + 1)
-        .map(normalizeSpacesAndTabs(_, opts))
-
-    // These are the lines from the first pasted line, going above
-    val prePastedLines = splitLines.take(startLine).reverse
+    val pastedLines = splitLines.slice(startLine, endLine + 1)
 
     // Do not adjust indentation if we pasted into an existing line content
     def pastedIntoNonEmptyLine = {
       pastedLines match {
-        case Array(singleLine) =>
-          val originalLine = singleLine.substring(0, originalStart) +
-            singleLine.substring(endPos.endColumn)
+        case array if array.length > 0 =>
```

Comment:
I changed the logic here to not reformat if we are printing into any line that already contains code.